semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/RemoveOrphanSemanticdbFiles.scala

```diff
@@ -12,30 +12,34 @@ import scala.meta.io.AbsolutePath
 object RemoveOrphanSemanticdbFiles {
 
   /** Removes *.semanticdb files that have no corresponding *.scala file. */
-  def process(sourceroot: AbsolutePath, targetroot: AbsolutePath): Unit = {
-    Files.walkFileTree(
-      targetroot.resolve(SemanticdbPaths.semanticdbPrefix).toNIO,
-      new SimpleFileVisitor[Path] {
-        override def visitFile(file: Path, attrs: BasicFileAttributes): FileVisitResult = {
-          if (PathIO.extension(file.getFileName) == SemanticdbPaths.semanticdbExtension) {
-            val scalafile = SemanticdbPaths.toScala(AbsolutePath(file), sourceroot, targetroot)
-            if (!scalafile.isFile) {
-              Files.delete(file)
+  def process(config: SemanticdbConfig): Unit = {
+    val sourceroot = config.sourceroot
+    val targetroot = config.targetroot
+    val semanticdbroot = targetroot.resolve(SemanticdbPaths.semanticdbPrefix).toNIO
+    if (Files.exists(semanticdbroot)) {
```

Comment:
nitpick, this is a moment where `if (!Files.exists(semanticdbroot)) return` might be nice to avoid the added indent 😉